src/p2/interpreter.rs

```diff
@@ -741,6 +766,38 @@ impl ParsedDocument {
         self.processing_imports = false;
     }
 
+    fn done_processing_terms(&mut self) {
+        self.processing_terms = false;
+    }
+
+    fn replace_terms_with_links(
+        &mut self,
+        _terms_map: &std::collections::HashMap<String, String>,
+    ) -> ftd::p1::Result<()> {
+        // Term syntax 1 = [term`](term: someTerm)
+        // Refer someTerm from linked text <term`>
+        // replacement = \[[term`]\]([document-id]#[slugified(someTerm)])
+        let _pattern_1 = regex::Regex::new(
+            r"(?x) # Enabling Comment Mode
+            \[(?P<linked_text>[\sa-zA-Z\d]+)\] # Linked Text Capture Group <linked_text>
+            \(\s*term\s*:(?P<actual_term>[\sa-zA-Z\d]+)\) # Referred Term Capture Group <actual_term>"
+        ).unwrap();
```

Comment:
You can compile regexes and store as lazy static so you can cache compilation output. 

Checkout: https://docs.rs/regex/1.6.0/regex/#example-avoid-compiling-the-same-regex-in-a-loop

Reply:
Do a search in entire code base where we have used regex without lazy static, and use this recommendation.